### ISSUE TITLE
build(common): add generate sample into a separate task

### ISSUE THREAD

**Issue by archerzz** [Azure/autorest.csharp#3067]
## Description

To reduce the impact, we should separate sample generation from code generation. And we can consider:
- Either we generate sample in a separate task inside the `Autorest C# publish` pipeline, in parallel with code generation
- Or we create a cron job to generate samples periodically
